homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -46,66 +60,106 @@
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    _state_map = {
+        YALE_STATE_DISARM: STATE_ALARM_DISARMED,
+        YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
+        YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
+    }
 
-        self._state_map = {
-            YALE_STATE_DISARM: STATE_ALARM_DISARMED,
-            YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
-            YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
-        }
+    _state = STATE_UNAVAILABLE
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self.coordinator.entry.data[CONF_NAME]
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return str(self.coordinator.entry.entry_id)  # type: ignore[attr-defined]
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return {
+            ATTR_NAME: self.coordinator.entry.data[CONF_NAME],  # type: ignore[attr-defined]
+            ATTR_MANUFACTURER: MANUFACTURER,
+            ATTR_MODEL: MODEL,
+            ATTR_IDENTIFIERS: {(DOMAIN, self.coordinator.entry.data[CONF_USERNAME])},  # type: ignore[attr-defined]
+        }
 
     @property
     def state(self):
         """Return the state of the device."""
         return self._state
 
+    @property
+    def code_arm_required(self):
+        """Whether the code is required for arm actions."""
+        return False
+
     @property
     def supported_features(self) -> int:
         """Return the list of supported features."""
         return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
 
-    def update(self):
-        """Return the state of the device."""
-        armed_status = self._client.get_armed_status()
-
-        self._state = self._state_map.get(armed_status)
-
     def alarm_disarm(self, code=None):
         """Send disarm command."""
-        self._client.disarm()
+        self.coordinator.yale.disarm()
 
     def alarm_arm_home(self, code=None):
         """Send arm home command."""
-        self._client.arm_partial()
+        self.coordinator.yale.arm_partial()
 
     def alarm_arm_away(self, code=None):
         """Send arm away command."""
-        self._client.arm_full()
+        self.coordinator.yale.arm_full()
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_CODE_FORMAT: self.code_format,
+            ATTR_CHANGED_BY: self.changed_by,
+            ATTR_CODE_ARM_REQUIRED: self.code_arm_required,
+            ATTR_ONLINE: self.coordinator.data["online"],
+            ATTR_STATUS: self.coordinator.data["status"],
+        }
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
```

Comment:
You can remove this entirely and let the `CoordinatorEntity` handle the update. Just return `self._state_map.get(self.coordinator.data["alarm"], STATE_UNAVAILABLE)` for the state property instead of `self._state`. The property will only be checked on state updates. You can also get rid of `self._state` entirely then

Reply:
Note that unavailable state should be set via the `available` property.

Reply:
Corrected as per suggestion of both